src/r1cs_nark_as/constraints/data_structures.rs

```diff
@@ -94,13 +72,27 @@ impl<CF: PrimeField> AllocVar<VerifierKey, CF> for VerifierKeyVar<CF> {
                 .collect::<Result<Vec<_>, SynthesisError>>()?;
 
             Ok(Self {
-                nark_index,
+                num_instance_variables,
+                hp_as_vk,
+                nark_matrices_hash,
                 as_matrices_hash,
             })
         })
     }
 }
 
+impl<CF: PrimeField> AbsorbableGadget<CF> for VerifierKeyVar<CF> {
+    fn to_sponge_field_elements(&self) -> Result<Vec<FpVar<CF>>, SynthesisError> {
+        let num_instance_variables = FpVar::Constant(CF::from(self.num_instance_variables as u128));
```

Comment:
```suggestion
        let num_instance_variables = FpVar::Constant(CF::from(self.num_instance_variables as u64));
```

Reply:
This is slightly more reasonable, given that all targets so far have |usize| < 64 bits